openxr/src/generated.rs

```diff
@@ -5216,6 +6395,140 @@ pub(crate) mod builder {
+                _marker: PhantomData,
+            }
+        }
+        #[inline]
+        pub fn into_raw(self) -> sys::HapticAmplitudeEnvelopeVibrationFB {
+            self.inner
+        }
+        #[inline]
+        pub fn as_raw(&self) -> &sys::HapticAmplitudeEnvelopeVibrationFB {
+            &self.inner
+        }
+        #[inline]
+        pub fn duration(mut self, value: Duration) -> Self {
+            self.inner.duration = value;
+            self
+        }
+        #[inline]
+        pub fn amplitudes(mut self, value: &'a [f32]) -> Self {
+            self.inner.amplitudes = value.as_ptr() as *const _ as _;
+            self.inner.amplitude_count = value.len() as u32;
+            self
+        }
+    }
+    impl<'a> Deref for HapticAmplitudeEnvelopeVibrationFB<'a> {
+        type Target = HapticBase<'a>;
+        #[inline]
+        fn deref(&self) -> &Self::Target {
+            unsafe { mem::transmute(&self.inner) }
+        }
+    }
+    impl<'a> Default for HapticAmplitudeEnvelopeVibrationFB<'a> {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+    #[derive(Copy, Clone)]
+    #[repr(transparent)]
+    pub struct HapticPcmVibrationFB<'a> {
+        inner: sys::HapticPcmVibrationFB,
+        _marker: PhantomData<&'a ()>,
+    }
+    impl<'a> HapticPcmVibrationFB<'a> {
+        #[inline]
+        pub fn new() -> Self {
+            Self {
+                inner: sys::HapticPcmVibrationFB {
+                    ty: sys::StructureType::HAPTIC_PCM_VIBRATION_FB,
+                    ..unsafe { mem::zeroed() }
+                },
+                _marker: PhantomData,
+            }
+        }
+        #[doc = r" Initialize with the supplied raw values"]
+        #[doc = r""]
+        #[doc = r" # Safety"]
+        #[doc = r""]
+        #[doc = r" The guarantees normally enforced by this builder (e.g. lifetimes) must be"]
+        #[doc = r" preserved."]
+        #[inline]
+        pub unsafe fn from_raw(inner: sys::HapticPcmVibrationFB) -> Self {
+            Self {
+                inner,
+                _marker: PhantomData,
+            }
+        }
+        #[inline]
+        pub fn into_raw(self) -> sys::HapticPcmVibrationFB {
+            self.inner
+        }
+        #[inline]
+        pub fn as_raw(&self) -> &sys::HapticPcmVibrationFB {
+            &self.inner
+        }
+        #[inline]
+        pub fn buffer(mut self, value: &'a [f32]) -> Self {
+            self.inner.buffer = value.as_ptr() as *const _ as _;
+            self.inner.buffer_size = value.len() as u32;
+            self
+        }
+        #[inline]
+        pub fn sample_rate(mut self, value: f32) -> Self {
+            self.inner.sample_rate = value;
+            self
+        }
+        #[inline]
+        pub fn append(mut self, value: bool) -> Self {
+            self.inner.append = value.into();
+            self
+        }
+        #[inline]
+        pub fn samples_consumed(mut self, value: &'a u32) -> Self {
```

Comment:
This value is modified by the runtime to report back the number of samples that were consumed. It would be great if this could be reflected by the type somehow. Are there any other examples of that?

Reply:
Good catch; in fact we *must* arrange for this to be `&mut`, or use will trigger UB. Can we select `&` vs `&mut` based on whether the pointee is `const`?

Reply:
I think that should work. Will try!

Reply:
Looks like it's working!

Reply:
Awesome! I guess this really was the first occurrence of an output parameter.